nifi-extension-bundles/nifi-kafka-bundle/nifi-kafka-service-shared/src/test/java/org/apache/nifi/kafka/service/verification/KafkaClusterVerifierTest.java

```diff
@@ -0,0 +1,276 @@
+
+    @BeforeEach
+    void setUp() {
+        socketConnector = (host, port, timeout) -> {
+            // Simulate successful connection by default
+        };
+        verifier = new KafkaClusterVerifier(VERIFY_TIMEOUT, logger, (host, port, timeout) -> socketConnector.accept(host, port, timeout));
+    }
+
+    @Test
+    void testVerifyClusterConnectivitySuccess() {
+        // Arrange
+        final Node node1 = new Node(1, "broker1.example.com", 9092);
+        final Collection<Node> nodes = List.of(node1);
+
+        final KafkaFuture<Collection<Node>> nodesFuture = KafkaFuture.completedFuture(nodes);
+        when(admin.describeCluster(any(DescribeClusterOptions.class))).thenReturn(describeClusterResult);
+        when(describeClusterResult.nodes()).thenReturn(nodesFuture);
+
+        // Act
+        final List<ConfigVerificationResult> results = verifier.verifyClusterConnectivity(admin, BOOTSTRAP_SERVERS);
+
+        // Assert - should have cluster description + 1 node result
+        assertEquals(2, results.size());
+
+        // Cluster description result
+        assertEquals(SUCCESSFUL, results.getFirst().getOutcome());
+        assertTrue(results.getFirst().getExplanation().contains("Cluster Nodes Found [1]"));
+
+        // Node verification result - check that we have at least one node result
+        final boolean hasNodeResult = results.stream()
+                .anyMatch(r -> r.getVerificationStepName().contains("Node 1")
+                    && r.getOutcome() == SUCCESSFUL
+                    && r.getVerificationStepName().contains("broker1.example.com"));
+        assertTrue(hasNodeResult, "Should have a node verification result");
+    }
+
+    @Test
+    void testVerifyClusterConnectivityWithAuthorizationException() {
+        // Arrange
+        when(admin.describeCluster(any(DescribeClusterOptions.class))).thenThrow(new AuthorizationException("Unauthorized"));
+
+        // Act
+        final List<ConfigVerificationResult> results = verifier.verifyClusterConnectivity(admin, BOOTSTRAP_SERVERS);
+
+        // Assert
+        assertFalse(results.isEmpty());
+
+        // Should have skipped cluster description
+        final ConfigVerificationResult clusterResult = results.stream()
+                .filter(r -> r.getVerificationStepName().contains("Cluster Description"))
+                .findFirst()
+                .orElseThrow();
+
+        assertEquals(SKIPPED, clusterResult.getOutcome());
+        assertTrue(clusterResult.getExplanation().contains("Insufficient permissions"));
+
+        // Should have bootstrap server results (fallback)
+        final long bootstrapResults = results.stream()
+                .filter(r -> r.getVerificationStepName().contains("Bootstrap Server Reachability"))
+                .count();
+
+        assertTrue(bootstrapResults > 0);
+
+        verify(logger).warn(eq("Describe Cluster insufficient permissions"), any(AuthorizationException.class));
```

Comment:
In general, logging messages should not be part of unit tests to avoid tight coupling to the implementation.